Backup / recovery documentation (#1377)
epiphany-platform/epiphany: docs/home/howto/BACKUP.md

#### Backup configuration

Copy default configuration for backup from ``defaults/configuration/backup.yml`` into newly created backup.yml config file, and enable backup for chosen components by setting up ``enabled`` parameter ``true``

This config may also be attached to cluster-config.yml

```
kind: configuration/backup
title: Backup Config
name: default
specification:
  components:
    load_balancer:
      enabled: true
    logging:
      enabled: false
    monitoring:
      enabled: true
    postgresql:
      enabled: true
    rabbitmq:
      enabled: false
#Kubernes backup and recovery is not supported by Epiphany.
#You may create backup by enabling this below, but recovery should be done manually according Kubernetes documentation.
    kubernetes:
      enabled: false
```

Run ``epicli backup`` command:
```
epicli backup -f backup.yml -b build_folder
```

If backup config is attached to cluster-config.yml use this file instead ``backup.yml``

## 2. How to store backup

Backup location is defined in ``backup`` role as ``backup_destination_host`` and ``backup_destination_dir``.
Default backup location is defined on ``repository`` host in location ``/epibackup/mounted/``.

> Use ``mounted`` location as mount point and mount storage you want to use. This might be:

Reviewer: ```suggestion
Use ``mounted`` location as mount point and mount there storage you want to use. This might be:
```

- Azure Blob Storage
- Amazon S3
- NAS
- Any other attached storage

Ensure that mounted location has enough space, is reliable and is well protected against disaster.

> ### If you don't attach any storage into the mount point location be aware that backups will be store on the local machine. This is not recommended.

Reviewer: ```suggestion
### If you don't attach any storage into the mount point location be aware that backups will be stored on the local machine. This is not recommended.
```

This config may also be attached to cluster-config.yml

```
kind: configuration/recovery
title: Recovery Config
name: default
specification:
  components:
    load_balancer:
      enabled: true
      snapshot_name: latest           #restore latest backup
    logging:
      enabled: true
      snapshot_name: 20200604-150829  #restore selected backup
    monitoring:
      enabled: false
      snapshot_name: latest
    postgresql:
      enabled: false
      snapshot_name: latest
    rabbitmq:
      enabled: false
      snapshot_name: latest
```

Run ``epicli recovery`` command:

``epicli recovery -f recovery.yml -b build_folder``

If recovery config is attached to cluster-config.yml use this file instead ``recovery.yml``

## 4. How backup and recovery work

### Load Balancer

Load balancer backup includes:
- Configuration files: ``/etc/haproxy/``
- SSL certificates: ``/etc/ssl/haproxy/``

> Recovery includes all backed up files

Reviewer: ```suggestion
Recovery includes all backed up files.
```

### Logging

Logging backup includes:
- Elasticsearch database snapshot
- Elasticsearch configuration ``/etc/elasticsearch/``
- Kibana configuration ``/etc/kibana/``

> Only single-node Elasticsearch backup is supported. Solution for multi-node Elasticsearch cluster will be added in future release.

Reviewer: ```suggestion
Only single-node Elasticsearch backup is supported. Solution for multi-node Elasticsearch cluster will be added in a future release.
```

### Monitoring
Monitoring backup includes:
- Prometheus data snapshot
- Prometheus configuration ``/etc/prometheus/``
- Grafana data snapshot

Recovery includes all backed up configurations and snapshots.

### Postgresql
Postgresql backup includes:
- Database data and metadata dump using ``pg_dumpall``
- Configuration files: ``*.conf``

> When multinode configuration is used, and faileover action has changed database cluster status (one node down, switchover) it's still possible to create backup. But before database restore, cluster needs to be recovered by run ``epicli apply`` and next ``epicli recovery`` to restore database data.

Reviewer: ```suggestion
When multinode configuration is used, and failover action has changed database cluster status (one node down, switchover) it's still possible to create backup. But before database restore, cluster needs to be recovered by running ``epicli apply`` and next ``epicli recovery`` restoring database data.
```

Author: i would stay with "to restore" - sounds better
